src/frontends/lean/parser.cpp

```diff
@@ -2092,21 +2123,28 @@ optional<expr> parser::resolve_local(name const & id, pos_info const & p, list<n
     // extra locals
     unsigned vidx = 0;
     for (name const & extra : extra_locals) {
-        if (id == extra)
-            return some_expr(save_pos(mk_var(vidx), p));
+        if (id == extra) {
+            expr r = save_pos(mk_var(vidx), p);
+            finalize_ast(id_data.m_id, r);
+            return some_expr(r);
+        }
         vidx++;
     }
 
     // locals
     if (auto it1 = m_local_decls.find(id)) {
-        return some_expr(copy_with_new_pos(*it1, p));
+        expr r = copy_with_new_pos(*it1, p);
+        finalize_ast(id_data.m_id, r);
+        return some_expr(r);
     }
 
     if (allow_field_notation && !id.is_atomic() && id.is_string()) {
-        if (auto r = resolve_local(id.get_prefix(), p, extra_locals)) {
+        ast_data prefix_data = id_data; // TODO(dselsam): implement
+        name n = id_data.m_value = id.get_prefix();
+        if (auto s = resolve_local(n, id_data, p, extra_locals)) {
             auto field_pos = p;
             field_pos.second += id.get_prefix().utf8_size();
-            return some_expr(save_pos(mk_field_notation_compact(*r, id.get_string()), field_pos));
+            return some_expr(save_pos(mk_field_notation_compact(*this, *s, field_pos, id.get_string()), field_pos));
```

Comment:
this calls `save_pos` twice

Reply:
good catch -- removed the save_pos at that line